scripts/submitArmadaSpark.sh

```diff
@@ -84,10 +88,25 @@ else
     )
 fi
 
+# Build deploy-mode specific arguments array
+DEPLOY_MODE_ARGS=()
+if [ "$DEPLOY_MODE" = "client" ]; then
+    DEPLOY_MODE_ARGS=(
+        --conf spark.driver.host=$SPARK_DRIVER_HOST
+        --conf spark.driver.port=$SPARK_DRIVER_PORT
+        --conf spark.app.id=armada-spark-$(openssl rand -hex 3)
```

Comment:
I'd really prefer the rand to happen inside our scala code.  The default should be random.  If the user wants to override for some reason, he can, but I don't want us to have to do this here.

Reply:
Made some changes to fix this.